packages/ai-jsx/src/experimental/next/chat.client.tsx

```diff
@@ -0,0 +1,143 @@
+/** @jsxImportSource ai-jsx/react */
+'use client';
+import React, { ReactNode, useContext, createContext } from 'react';
+import { JSX } from 'ai-jsx/react/jsx-runtime';
+
+/**
+ * UI that represents the contents of a message.
+ */
+export interface UIMessage {
+  type: 'assistant' | 'user';
+  message: ReactNode;
+}
+
+/**
+ * A Server Action that processes a message and returns the resulting UI and context for the next message.
+ * See the `sendMessage` helper in `ai-jsx/experimental/next/chat`.
+ */
+export type SendMessage<T> = (formData: FormData, context: T | undefined) => Promise<[UIMessage[], Promise<T>]>;
+
+/**
+ * React Context containing the `UIMessage`s associated with a conversation. Typically consumed indirectly via
+ * the `<Conversation>` component.
+ */
+export const ConversationContext = createContext([] as UIMessage[]);
+
+/**
+ * React Context containing the UI of the contents of a single message. Typically consumed indirectly via the
+ * `<Message>` component.
+ */
+export const MessageContext = createContext(null as ReactNode);
+
+/**
+ * React Context containing the form status. Typically consumed indirectly via the `<ChatState>` component.
+ */
+export const StatusContext = createContext<'idle' | 'pending' | 'streaming'>('idle');
```

Comment:
Minor, but if we're using `'idle' | 'pending' | 'streaming'` in multiple places, it would be nice to extract into a helper type.